monitoring/sockets.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"context"
+
+	"github.com/gravitational/satellite/agent/health"
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+	"github.com/gravitational/trace"
+)
+
+// NewSocketsChecker creates and returns new SocketChecker
+func NewSocketsChecker(tcpPorts []int, udpPorts []int, unixSocks []string) health.Checker {
+	c := &SocketsChecker{
+		TCPPorts:    make([]int, len(tcpPorts)),
+		UDPPorts:    make([]int, len(udpPorts)),
+		UNIXSockets: make([]string, len(unixSocks)),
+	}
+	copy(c.TCPPorts, tcpPorts)
+	copy(c.UDPPorts, udpPorts)
+	copy(c.UNIXSockets, unixSocks)
+	return c
+}
+
+// SocketsChecker is a health.Checker which is able to check if specified TCP/UDP/Unix sockets are open on a host
+type SocketsChecker struct {
+	TCPPorts    []int
+	UDPPorts    []int
+	UNIXSockets []string
+}
+
+// Name returns name of this checker
+func (c *SocketsChecker) Name() string {
+	return "sockets"
+}
+
+// checkTCPPorts checks if specified slice of TCP ports are opened and returns slice of errors found
+func checkTCPPorts(ports []int) []error {
+	errors := make([]error, 0)
+	if len(ports) == 0 {
+		return errors
```

Comment:
```go
return nil
```
as you don't need to pre-allocate slices (as opposed to maps - which you do need to pre-allocate if you wanted to return a value that can be used for appending w/o having to check for `nil` first).